Fails to build on Darwin.

> ```
> make  install-exec-hook
> make[2]: Entering directory '/private/tmp/nix-build-tnat64-0.05.drv-0/source'
> cd /nix/store/yhxvz2pc9md03cbp0d4gsiw1w6psmrcp-tnat64-0.05/lib/tnat64 && \
>  rm -f libtnat64.a && \
>  rm -f libtnat64.la && \
>  rm -f libtnat64.so && \
>  rm -f libtnat64.so.0 && \
>  mv libtnat64.so.0.0.0 libtnat64.so
> mv: cannot stat 'libtnat64.so.0.0.0': No such file or directory
> make[2]: *** [Makefile:1218: install-exec-hook] Error 1
> ```

Please patch [that hook](https://github.com/andrewshadura/tnat64/blob/cfb19ac2e77287768e922244e8c2cb57a812f17e/Makefile.am#L41-L47) to work on Darwin (`libtnat64.0.dylib` is the longest variant I can find in the logs), patch it away altogether or mark the package broken on Darwin.

--- a/pkgs/tools/networking/tnat64/default.nix
+++ b/pkgs/tools/networking/tnat64/default.nix
@@ -1,0 +1,37 @@
+{ lib, stdenv, fetchFromGitHub, autoreconfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "tnat64";
+  version = "0.05";
+
+  src = fetchFromGitHub {
+    owner = "andrewshadura";
+    repo = pname;
+    rev = "${pname}-${version}";
+    sha256 = "07lmzidbrd3aahk2jvv93cic9gf36pwmgfd63gmy6hjkxf9a6fw9";
+  };
+
+  postPatch = ''
+    # Fix usage of deprecated sys_errlist
+    substituteInPlace tnat64.c --replace 'sys_errlist[errno]' 'strerror(errno)'
+  '';
+
+  configureFlags = [ "--libdir=$(out)/lib" ];
+  nativeBuildInputs = [ autoreconfHook ];
+
+  meta = with lib; {
+    description = "IPv4 to IPv6 interceptor";
+    homepage = "https://github.com/andrewshadura/tnat64";
+    license = licenses.gpl2Plus;
+    longDescription = ''
+      TNAT64 is an interceptor which redirects outgoing TCPv4 connections
+      through NAT64, thus enabling an application running on an IPv6-only host
+      to communicate with the IPv4 world, even if that application does not
+      support IPv6 at all.
+    '';
+    platforms = platforms.unix;
+    badPlatforms = platforms.darwin;
+    maintainers = [ maintainers.rnhmjoj ];
+  };
+
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -810,6 +810,8 @@ in
 
   tfk8s = callPackage ../tools/misc/tfk8s { };
 
+  tnat64 = callPackage ../tools/networking/tnat64 { };
+
   xcd = callPackage ../tools/misc/xcd { };
 
   xtrt = callPackage ../tools/archivers/xtrt { };

